src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -98,6 +128,22 @@ public class JsonInputFormatter : TextInputFormatter
             }
 
             var request = context.HttpContext.Request;
+
+            if (request.Body.CanSeek)
+            {
+                request.Body.Seek(0L, SeekOrigin.Begin);
+            }
+            else if (!_suppressInputFormatterBuffering)
+            {
+                // JSON.Net does synchronous reads. In order to avoid blocking on the stream, we asynchronously 
+                // read everything into a buffer, and then seek back to the beginning. 
+                BufferingHelper.EnableRewind(request);
+                Debug.Assert(request.Body.CanSeek);
+
+                await request.Body.DrainAsync(context.HttpContext.RequestAborted);
```

Comment:
I don't think we should pass a cancellation token here. MVC doesn't have any other behavior where we cancel based on request aborted.